src/CommandLineUtils/CommandOption.cs

```diff
@@ -132,6 +132,11 @@ internal CommandOption(CommandOptionType type)
         /// </summary>
         public bool Inherited { get; set; }
 
+        /// <summary>
+        /// Defines the underlying type of the option for the help-text-generator
+        /// </summary>
+        public Type UnderlyingType { get; set; }
```

Comment:
```suggestion
        internal Type UnderlyingType { get; set; }
```

I think this property would best be left as internal. I see this as an implementation  detail that should be hidden. I could see this changing in the future and don’t want to add an API if we don’t have to.

Reply:
Absolutely not!. I spent so much time figuring out how to introduce the bloody public's in the first place! ;-)